tests/v2/test_waku_store.nim

```diff
@@ -377,3 +377,84 @@ procSuite "Waku Store":
+        WakuMessage(payload: @[byte 2],contentTopic: ContentTopic(2), timestamp: float(2)),
+        WakuMessage(payload: @[byte 3],contentTopic: ContentTopic(1), timestamp: float(3)),
+        WakuMessage(payload: @[byte 4],contentTopic: ContentTopic(2), timestamp: float(4)),
+        WakuMessage(payload: @[byte 5],contentTopic: ContentTopic(1), timestamp: float(5)),
+        WakuMessage(payload: @[byte 6],contentTopic: ContentTopic(2), timestamp: float(6)),
+        WakuMessage(payload: @[byte 7],contentTopic: ContentTopic(1), timestamp: float(7)),
+        WakuMessage(payload: @[byte 8],contentTopic: ContentTopic(2), timestamp: float(8)),
+        WakuMessage(payload: @[byte 9],contentTopic: ContentTopic(1),timestamp: float(9))]
+            
+    var dialSwitch = newStandardSwitch()
+    discard await dialSwitch.start()
+
+    var listenSwitch = newStandardSwitch(some(key))
+    discard await listenSwitch.start()
+
+    let
+      proto = WakuStore.init(PeerManager.new(dialSwitch), crypto.newRng())
+      subscription = proto.subscription()
+    proto.setPeer(listenSwitch.peerInfo)
+
+    var subscriptions = newTable[string, MessageNotificationSubscription]()
+    subscriptions["test"] = subscription
+
+    listenSwitch.mount(proto)
+
+    for wakuMsg in msgList:
+      await subscriptions.notify("foo", wakuMsg)
+    
+    asyncTest "handle temporal history query with a valid time window":
+      var completionFut = newFuture[bool]()
+
+      proc handler(response: HistoryResponse) {.gcsafe, closure.} =
+        check:
+          response.messages.len() == 2
+          response.messages.anyIt(it.timestamp == float(3))
+          response.messages.anyIt(it.timestamp == float(5))
+        completionFut.complete(true)
+
+      let rpc = HistoryQuery(topics: @[ContentTopic(1)], startTime: float(2), endTime: float(5))
+      await proto.query(rpc, handler)
+
+      check:
+        (await completionFut.withTimeout(5.seconds)) == true
+
+    asyncTest "handle temporal history query with a zero-size time window":
+      var completionFut = newFuture[bool]()
+
+      proc handler(response: HistoryResponse) {.gcsafe, closure.} =
+        check:
+          response.messages.len() == 5
+        completionFut.complete(true)
+
+      let rpc = HistoryQuery(topics: @[ContentTopic(1)], startTime: float(2), endTime: float(2))
+      await proto.query(rpc, handler)
+
+      check:
+        (await completionFut.withTimeout(5.seconds)) == true
+
+    asyncTest "handle temporal history query with an invalid time window":
+      var completionFut = newFuture[bool]()
+
+      proc handler(response: HistoryResponse) {.gcsafe, closure.} =
+        check:
+          response.messages.len() == 5
+        completionFut.complete(true)
+
+      # time window is invalid since start time > end time
+      let rpc = HistoryQuery(topics: @[ContentTopic(1)], startTime: float(5), endTime: float(2))
```

Comment:
My first thought would be that a query with an explicit but invalid time range should return zero messages rather than quietly ignoring the time arguments and returning everything?

Reply:
Thanks @jm-clius! Oskar also had a similar comment, please see my answer https://github.com/status-im/nim-waku/pull/454#discussion_r608838867